extensions/chatgpt/src/views/vision.tsx

```diff
@@ -8,13 +8,17 @@ import { toUnit } from "../utils";
 import { LoadFrom, loadFromClipboard, loadFromFinder } from "../utils/load";
 import { countImageTokens, countToken, estimateImagePrice, estimatePrice } from "../utils/token";
 
+const preferences = getPreferenceValues<Preferences>();
+
+const visionModelName: string = preferences.useVisionModel ? preferences.visionModelName ?? "gpt-4o" : "gpt-4o";
```

Comment:
I was a bit confused about this line, is it the same as

```ts
const visionModelName: string = preferences.useVisionModel && preferences.visionModelName || "gpt-4o";
```

Reply:
Sorry for the confusion. 
This code, is used to determine whether to use the `gpt4-o` model or a custom model. 

> If `useVisionModel` is true and `visionModelName` has a specific value, then the custom `visionModelName` model is used. Otherwise, it's using the `gpt4-o` model. 

BTW, your code is fine and better than mine. I'm actually a business major and was looking forward to this feature for a long time, but never saw it. So I wanted to try it myself. But I'm not very familiar with programming, that's why I wrote this "weird" code.